source/features/preserve-text-messages.js

```diff
@@ -0,0 +1,116 @@
+import debounce from 'lodash.debounce';
+import {observeEl} from '../libs/utils';
+
+let DMDialogOpen = false;
+
+function onDMDialogOpen() {
+	observeEl('#dm_dialog', async mutations => {
+		for (const mutation of mutations) {
+			if (mutation.target.style.display === 'none') {
+				DMDialogOpen = false;
+			} else {
+				DMDialogOpen = true;
+				onDMOpen();
+			}
+		}
+
+		if (!DMDialogOpen) {
+			const savedMessages = await browser.storage.local.get();
+			const idsOfMessagesToRemove = [];
+
+			for (const id in savedMessages) {
+				if (isEmptyMsgInput(savedMessages[id])) {
```

Comment:
This could also be a `filter`, specially since you have the function done already